app/Models/Collection.php

```diff
@@ -466,4 +466,13 @@ public function isBlacklisted(): bool
     {
         return BlacklistedCollections::includes($this->address);
     }
+
+    /**
+     * @param  Builder<self>  $query
+     * @return Builder<self>
+     */
+    public function scopeWithSignedWallet(Builder $query): Builder
+    {
+        return $query->whereHas('nfts', fn ($nftQuery) => $nftQuery->whereHas('wallet', fn ($walletQuery) => $walletQuery->hasBeenSigned()));
```

Comment:
Hmm, for very large collections this might be very slow. Worst case we have to scan each nft in each collection.

Maybe alternatively we can use a special purpose query.

For example, something like this:

```sql
with signed_wallets as (
    select wallet_id from wallets
    -- should be indexed, even better if it has a separate column because it's simpler to follow
    where extra_attributes->last_signed_at is not null 
), collection_ids as (
    select distinct collection_id from nfts
    where nfts.wallet_id in (select wallet_id from signed_wallets)
)
select * from collections
where in in (select collection_id from collection_ids)
order by ...
limit ...
```

(haven't tested it, but you get the idea)

Reply:
see discussion on slack @alfonsobries